lib/slide_hub/cloud/engine/aws.rb

```diff
@@ -0,0 +1,280 @@
+          include ActiveSupport::Configurable
+          config_accessor :bucket_name
+          config_accessor :image_bucket_name
+          config_accessor :use_s3_static_hosting
+          config_accessor :region
+          config_accessor :cdn_base_url
+          config_accessor :sqs_url
+          config_accessor :aws_access_id
+          config_accessor :aws_secret_key
+        end
+
+        def self.configure(&block)
+          yield config
+
+          if self.configured?
+            Aws.config.update({
+              region: @config.region,
+              credentials: Aws::Credentials.new(@config.aws_access_id, @config.aws_secret_key),
+            },)
+          end
+        end
+
+        def self.config
+          @config ||= Config.new
+        end
+
+        def self.configured?
+          return false unless defined? @config.aws_access_id
+          return false unless defined? @config.aws_secret_key
+          return false if @config.aws_access_id.blank?
+          return false if @config.aws_secret_key.blank?
+          true
+        end
+
+        def self.resource_endpoint
+          return @config.cdn_base_url unless @config.cdn_base_url.blank?
+
+          url = if @config.use_s3_static_hosting == '1'
+                  "http://#{@config.image_bucket_name}"
+                elsif @config.region == 'us-east-1'
+                  "https://#{@config.image_bucket_name}.s3.amazonaws.com"
+                else
+                  "https://#{@config.image_bucket_name}.s3-#{@config.region}.amazonaws.com"
+                end
+          url
+        end
+
+        def self.upload_endpoint
+          url = if @config.region == 'us-east-1'
+                  "https://#{@config.bucket_name}.s3.amazonaws.com"
+                else
+                  "https://#{@config.bucket_name}.s3-#{@config.region}.amazonaws.com"
+                end
+          url
+        end
+
+        def self.s3_host_name
+          s3_host_name = if @config.region == 'us-east-1'
+                           's3.amazonaws.com'
+                         else
+                           "s3-#{@config.region}.amazonaws.com"
+                         end
+          s3_host_name
+        end
+
+        ## SQS
+        def self.sqs
+          @sqs ||= Aws::SQS::Client.new(region: @config.region)
+        end
+
+        def self.send_message(message)
+          resp = self.sqs.send_message({
+            queue_url: @config.sqs_url,
+            message_body: message,
+          },)
+          resp
+        end
+
+        def self.receive_message(max_number = 10)
+          response = self.sqs.receive_message(queue_url: @config.sqs_url, visibility_timeout: 600, max_number_of_messages: max_number)
+          # see http://docs.aws.amazon.com/sdkforruby/api/Aws/SQS/Client.html#receive_message-instance_method
+          result = SlideHub::Cloud::Queue::Response.new
+          unless (!response || !response.respond_to?(:messages) || response.messages.count == 0)
```

Comment:
__[rubocop]__
* Style/ParenthesesAroundCondition: Don't use parentheses around the condition of an `unless`. [:ref](https://github.com/bbatsov/ruby-style-guide#no-parens-if)